### ISSUE TITLE
Kitten Cannon cat is fired out of the top of the screen

### ISSUE THREAD

**Issue by parkerlreed** [ruffle-rs/ruffle#1635]
Seems to affect trajectory as well. I can be aiming low and the cat will drop onto a spike only a couple hundred feet ahead.

Ruffle 0.1.0-nightly (c60dd208 2020-11-20)

https://streamable.com/9409fs

SWF from Flashpoint http://infinity.unstable.life/Flashpoint/Legacy/htdocs/localflash/kittencannon.swf

Also scores >= 1000 aren't rendered

![image](https://user-images.githubusercontent.com/841440/99872434-c20f8400-2baf-11eb-907a-5aa28c416ca5.png)

**Comment by Toad06** [comment]
Thanks for your bug report.

The main issue is that `new Object()` doesn't return an expected value here (cc #1469), so the value passed to `localToGlobal` isn't correct.

For the text thing, the game does `arr = arr.reverse();` but this method returns "undefined" in Ruffle:
```
var a = new Array(1,2,3);
trace(a); // "1, 2, 3"
a = a.reverse();
trace(a); "3, 2, 1" in Flash, "undefined" in Ruffle
```

**Comment by parkerlreed** [comment]
Score display fix works!

![image](https://user-images.githubusercontent.com/841440/99917624-79ef7f00-2cdf-11eb-83dc-c127464b20b7.png)

**Comment by can3p** [comment]
Just curious: I've checked actionscript 2 language reference and the signature of `reverse` method is `public reverse() : Void`, which means previous ruffle behaviour was a correct one. Does the actual flash player diverge from spec there?

**Comment by Herschel** [comment]
Yes, apparently the docs are mistaken and `Array.reverse` returns the modified array.

**Comment by parkerlreed** [comment]
@Herschel just tested that branch. Works great for Kitten Cannon!

**Comment by Herschel** [comment]
Fixed by #2415.
